templates/definition/meter/bge_tech_ws100.yaml

```diff
@@ -0,0 +1,73 @@
+template: bge_tech_ws100
+products:
+  - brand: BGEtech
+    description:
+      generic: WS100
+params:
+  - name: usage
+    choice: ["grid", "pv", "charge", "aux"]
+  - name: modbus
+    choice: ["rs485", "tcpip"]
+    baudrate: 1200
+render: |
+  type: custom
+  power:
+    source: modbus
+    {{- include "modbus" . | indent 2 }}
+    register:
+      address: 0x0104 # Total active power
+      type: holding
+      decode: int32
+    {{- if eq .usage "pv" }}
+    scale: -1
+    {{- end }}
+  energy:
+    source: modbus
+    {{- include "modbus" . | indent 2 }}
+    register:
+      {{- if eq .usage "pv" }}
+      address: 0x0118 # Reversing active energy consumption
+      {{- else }}
+      address: 0x010E # Forward active energy consumption
+      {{- end }}
+      type: holding
+      decode: int32
+    scale: 0.01
+  currents:
+  - source: modbus
+    {{- include "modbus" . | indent 2 }}
+    register:
+      address: 0x0102 # Electricity of A phase
+      type: holding
+      decode: int32
+    scale: 0.001
+  - source: const
+    value: 0  # since the device only a single phase, this value is set to 0
+  - source: const
+    value: 0  # since the device only a single phase, this value is set to 0
+  voltages:
+  - source: modbus
+    {{- include "modbus" . | indent 2 }}
+    register:
+      address: 0x0100 # Voltage of A phase
+      type: holding
+      decode: int32
+    scale: 0.001
+  - source: const
+    value: 0  # since the device only a single phase, this value is set to 0
+  - source: const
+    value: 0  # since the device only a single phase, this value is set to 0
+  powers:
+  - source: modbus
+    {{- include "modbus" . | indent 2 }}
+    register:
+      address: 0x0104 # Active power of A phase
+      type: holding
+      decode: int32
+    {{- if eq .usage "pv" }}
+    scale: -1
+    {{- end }}
+  - source: const
+    value: 0  # since the device only a single phase, this value is set to 0
```

Comment:
@premultiply I'm not happy with this pattern but also don't have a (much) better idea. Wdyt?